src/VecSim/algorithms/brute_force/bf_batch_iterator.cpp

```diff
@@ -25,6 +25,36 @@ VecSimQueryResult *BF_BatchIterator::searchByHeuristics(size_t n_res,
     return res;
 }
 
+void BF_BatchIterator::swapScores(const unordered_map<size_t, size_t> &TopCandidatesIndices,
+                                  size_t res_num) {
+    // Create a set of the indices in the scores array for every results that we return.
+    set<size_t> indices;
```

Comment:
should it be sorted set? in order to enforce right order of iteration when you compare `it` to `ind`

Reply:
As far as I know `set` in c++ is the basically a sorted set. `unsorted_set` is a set which is implemented with hash table...